### PR TITLE
Update project for Xcode 9.1, add codesigning flags

### PR DESCRIPTION
**Pull request by russellhancox**

### CODE REVIEW

File: Source/santad/DataLayer/SNTRuleTable.m [hunk 1]
```diff
@@ -133,7 +133,8 @@ - (SNTRule *)ruleForBinarySHA256:(NSString *)binarySHA256
 
 #pragma mark Adding
 
-- (BOOL)addRules:(NSArray *)rules cleanSlate:(BOOL)cleanSlate error:(NSError **)error {
+- (BOOL)addRules:(NSArray *)rules cleanSlate:(BOOL)cleanSlate
+           error:(NSError * __autoreleasing *)error {
```
Comment (tburgin) [inline Source/santad/DataLayer/SNTRuleTable.m]: I would double check this doesn't release too early.

File: Santa.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -2014,13 +2038,37 @@
...
 				GCC_OPTIMIZATION_LEVEL = fast;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
 				LLVM_LTO = YES;
 				MACOSX_DEPLOYMENT_TARGET = 10.9;
+				OTHER_CODE_SIGN_FLAGS = "-o library,kill";
```
Comment (tburgin) [inline Santa.xcodeproj/project.pbxproj]: "kill" is compatible for all the mach-os we compile?

File: Source/santad/DataLayer/SNTRuleTable.m (hunk 1 above)
Comment (russellhancox) [reply Source/santad/DataLayer/SNTRuleTable.m]: It doesn't, this is the fix recommended by clang (see [here](http://lists.llvm.org/pipermail/cfe-commits/Week-of-Mon-20170213/185409.html))

File: Santa.xcodeproj/project.pbxproj (hunk 2 above)
Comment (russellhancox) [reply Santa.xcodeproj/project.pbxproj]: As far as I know yes. I've also removed these flags from debug builds as it doesn't play nice with asan.

**Pull request merged**